lib/exercises.rb

```diff
@@ -1,29 +1,163 @@
+require 'set'
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
 
-def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+# Time Complexity: O(n) where n is the total number of characters in all strings
+# >> Go through each character in every string by visiting each string in array
+# and each character in each string
+# >> O(1) for ord
+# >> O(k), where k == num strings in input array, to get all hash values
+
+# Space Complexity: O(n)
+
+# Notes:
+# >> considered just assigning score using ord,
+# but this doesn't account for diff combinations of letters that sum to same score
+# >> FIX:
+# >> keep track of individual alphabet chars using array w/ 26 indices
+# >> index into that array using ch.ord - 'a'.ord (to baseline 'a' to 0)
+
+def grouped_anagrams(strings_arr)
```

Comment:
This is a quite clever solution, you didn't even need to sort the letters and use them as the key.